src/Microsoft.TemplateEngine.Core/Expressions/VisualBasic/VisualBasicStyleEvaluatorDefintion.cs

```diff
@@ -0,0 +1,357 @@
+using System.Globalization;
+using System.IO;
+using System.Text;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Core.Contracts;
+using Microsoft.TemplateEngine.Core.Util;
+
+namespace Microsoft.TemplateEngine.Core.Expressions.VisualBasic
+{
+    public class VisualBasicStyleEvaluatorDefintion
+    {
+        private static readonly IOperatorMap<Operators, Tokens> Map = new OperatorSetBuilder<Tokens>(Encode, Decode)
+            .And(Tokens.And)
+            .And(Tokens.AndAlso)
+            .Or(Tokens.Or)
+            .Or(Tokens.OrElse)
+            .Not(Tokens.Not)
+            .Xor(Tokens.Xor)
+            .GreaterThan(Tokens.GreaterThan, evaluate: (x, y) => Compare(x, y) > 0)
+            .GreaterThanOrEqualTo(Tokens.GreaterThanOrEqualTo, evaluate: (x, y) => Compare(x, y) >= 0)
+            .LessThan(Tokens.LessThan, evaluate: (x, y) => Compare(x, y) < 0)
+            .LessThanOrEqualTo(Tokens.LessThanOrEqualTo, evaluate: (x, y) => Compare(x, y) <= 0)
+            .EqualTo(Tokens.EqualTo, evaluate: (x, y) => Compare(x, y) == 0)
+            .NotEqualTo(Tokens.NotEqualTo, evaluate: (x, y) => Compare(x, y) != 0)
+            .Ignore(Tokens.Space, Tokens.Tab)
+            .LiteralBoundsMarkers(Tokens.Quote)
+            .OpenGroup(Tokens.OpenBrace)
+            .CloseGroup(Tokens.CloseBrace)
+            .TerminateWith(Tokens.WindowsEOL, Tokens.UnixEOL, Tokens.LegacyMacEOL)
+            .LeftShift(Tokens.LeftShift)
+            .RightShift(Tokens.RightShift)
+            .Add(Tokens.Add)
+            .Subtract(Tokens.Subtract)
+            .Multiply(Tokens.Multiply)
+            .Divide(Tokens.Divide)
+            .Exponentiate(Tokens.Exponentiate)
+            .Literal(Tokens.Literal)
+            .LiteralBoundsMarkers(Tokens.DoubleQuote)
+            .TypeConverter<VisualBasicStyleEvaluatorDefintion>(ConfigureConverters);
+
+        private static readonly IOperationProvider[] NoOperationProviders = new IOperationProvider[0];
+
+        private static readonly Dictionary<Encoding, ITokenTrie> TokenCache = new Dictionary<Encoding, ITokenTrie>();
+
+        private enum Tokens
+        {
+            And = 0,
+            AndAlso = 1,
+            Or = 2,
+            OrElse = 3,
+            Not = 4,
+            GreaterThan = 5,
+            GreaterThanOrEqualTo = 6,
+            LessThan = 7,
+            LessThanOrEqualTo = 8,
+            EqualTo = 9,
+            NotEqualTo = 10,
+            Xor = 11,
+            OpenBrace = 12,
+            CloseBrace = 13,
+            Space = 14,
+            Tab = 15,
+            WindowsEOL = 16,
+            UnixEOL = 17,
+            LegacyMacEOL = 18,
+            Quote = 19,
+            LeftShift = 20,
+            RightShift = 21,
+            Add = 22,
+            Subtract = 23,
+            Multiply = 24,
+            Divide = 25,
+            Exponentiate = 26,
+            DoubleQuote = 27,
+            Literal = 28,
+        }
+
```

Comment:
Many methods herein are exact copies of their Cpp2StyleEvaluatorDefinition counterparts. Make them shared?
- Evaluate()
- EvaluateFromString()
- AttemptNumericComparison()
- AttemptBooleanComparison()
- AttemptVersionComparison()
- AttemptLexographicComparison()
- AttemptComparableComparison()